File: backend/app/schemas/evidence.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import Optional
from uuid import UUID

class EvidenceBase(BaseModel):
    original_filename: str
    uploaded_by: str

class EvidenceCreate(EvidenceBase):
    pass

class EvidenceResponse(BaseModel):
    id: UUID
    evidence_id: str
    original_filename: str
    mime_type: Optional[str]
    file_size: int
    sha256: str
    uploaded_by: str
    uploaded_at: datetime
    verification_status: str
    wrapped_dek: str
    storage_provider: str
    storage_ref: str
    encryption_version: int
    encrypted_file_size: Optional[int]
    case_id: Optional[UUID]

    model_config = ConfigDict(from_attributes=True)

class EvidenceListResponse(BaseModel):
    total: int
    page: int
    page_size: int
    items: list[EvidenceResponse]

class VerificationResponse(BaseModel):
    verified: bool
    status: str
    message: str
    original_hash: Optional[str] = None
    current_hash: Optional[str] = None

File: src/types/evidence.ts
export interface Evidence {
  id: string; // UUID
  evidence_id: string;
  original_filename: string;
  mime_type: string;
  file_size: number;
  sha256: string;
  uploaded_by: string;
  uploaded_at: string;
  verification_status: 'pending' | 'verified' | 'tampered';
  wrapped_dek: string;
  storage_provider: string;
  storage_ref: string;
  encryption_version: number;
  encrypted_file_size: number;
  case_id?: string; // UUID
}

export interface CustodyEvent {
  id: number;
  evidence_id?: string;
  case_id?: string;
  event_type: string;
  actor: string;
  timestamp: string;
  description: string;
  metadata_json?: any;
  previous_event_hash?: string;
  event_hash: string;
}

export interface VerificationResponse {
  verified: boolean;
  status: 'verified' | 'tampered';
  original_hash: string;
  current_hash: string;
  message: string;
}

export interface EvidenceListResponse {
  total: number;
  page: number;
  page_size: number;
  items: Evidence[];
}
